src/shared/navgen/nav.cpp

```diff
@@ -1009,16 +1037,49 @@ static NavgenStatus rasterizeTileLayers( Geometry& geo, rcContext &context, int
 	return {};
 }
 
-void NavmeshGenerator::StartGeneration( class_t species )
+void NavmeshGenerator::EnqueueTasks( Str::StringRef mapName, std::bitset<PCL_NUM_CLASSES> classes )
+{
+	// The NavmeshGenerator object is not designed to be used more than once
+	ASSERT( mapName_.empty() );
+
+	// never divide by 0
+	// This fails to include map loading time but we don't update progress during that anyway
+	int amountOfWork = 1;
+
+	std::string names;
+
+	for ( int i = PCL_NUM_CLASSES; --i != PCL_NONE; )
+	{
+		if ( !classes[ i ] )
+		{
+			continue;
+		}
+
+		Init( mapName );
```

Comment:
Just init once outside the main loop to make the intent clear?

Reply:
Done.